libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -295,7 +295,7 @@ void GCS_MAVLINK::ftp_worker(void) {
                         put_le32_ptr(reply.data, (uint32_t)file_size);
 
                         // provide compatibility with old protocol banner download
-                        if (strncmp((const char *)request.data, "@PARAM/param.pck", 16) == 0) {
+                        if (strlen((const char *)request.data) == 16 && (strncmp((const char *)request.data, "@PARAM/param.pck", 16) == 0)) {
```

Comment:
```suggestion
                        if (strnlen((const char *)request.data, ARRAY_SIZE(request.data)) == 16 && (strncmp((const char *)request.data, "@PARAM/param.pck", 16) == 0)) {
```

This is a good fix.  But we don't want to over-read `request.data` if we've been passed garbage in.